### ISSUE TITLE
Question: Should we avoid having Partial field inside Partial object

### ISSUE THREAD

**Issue by atschabu** [IZIVIA/ocpi-toolkit#34]
As far as I can see, all generated Partials, use Partials as field values. For example `SessionPartial.charging_periods` is of type `List<ChargingPeriodPartial>?`, but the expectation is that we provide a valid ChargingPeriod, not partially updating existing ChargingPeriods (as that would require a PUT Session call)

> When a PATCH request contains the charging_periods field (inside a [Session](https://github.com/ocpi/ocpi/blob/master/mod_sessions.asciidoc#mod_sessions_session_object) object), this SHALL be processed as a request to add all the [ChargingPeriod](https://github.com/ocpi/ocpi/blob/master/mod_cdrs.asciidoc#mod_cdrs_chargingperiod_class) objects to the existing [Session](https://github.com/ocpi/ocpi/blob/master/mod_sessions.asciidoc#mod_sessions_session_object) object.

The issue is worse with the Location object, where we have about 10+ fields that are complex objects. Unless I interpret the protocol incorrectly, only the provided root object can be Partial, while all field values should be "regular" fully valid objects. Therefore types like `GeoLocation` and `BusinessDetail` don't even need to be annotated as `@Partial`, and generated Partials should not use Partial fields as field types. Which means a LocationPartial should contain a list of Evse, and only when calling PATCH Evse endpoint should we using EvsePartial, etc.

**Comment by lilgallon** [comment]
Yes I think you are right, I recoded generation logic here: https://github.com/IZIVIA/ocpi-toolkit/pull/31, and I can make the changes you are requesting there

**Comment by atschabu** [comment]
it looks like "it depends" on how one interprets the protocol ... I would have assumed that a change to an evse will always use the PATCH evse endpoint, but apparently that is not common consensus given how the creator of this issue experienced a partner wanting to use the PATCH location endpoint with a wonky json: https://github.com/ocpi/ocpi/issues/285

assuming you have any impact on the partners implementation of the PATCH endpoint, and can make them only send PATCH for root element, no need for nested Partials ... on the other hand, if you need to be able to consume a crazy json merge object like the one in 285, you'd need nested Partials
